app/controllers/StudentControllerAJAX.php

```diff
@@ -0,0 +1,43 @@
+<?php
+    require_once($GLOBALS['dirpre'].'controllers/Controller.php');
+
+    interface StudentControllerAJAXInterface {
+      /**
+       * Calls made when viewing a job/sublet
+       */
+      public static function bookmark();
+    }
+
+    class StudentControllerAJAX
+      extends StudentController
+      implements StudentControllerAJAXInterface {
+
+      // AJAX controller for students
+
+      // Takes POST data from job/sublet page and adds it to student's bookmarks
+      // $_POST will contain: id, type, title
+      public static function bookmark() {
```

Comment:
Should rename this (and route) to like addBookmark, and should have function for removing bookmarks and getting a list of bookmarks in json form as well.